### ISSUE TITLE
pod readiness check should not require port forwarding

### ISSUE THREAD

**Issue by errordeveloper** [weaveworks/eksctl#1228]
It is the job of Kubernetes to tell us whether a pod is ready or not; we don't need to setup a tunnel and call on pod health endpoint to check the readiness, Kubernetes does that for us.

We happen to have rather elaborate logic with multiple failure modes.

https://github.com/weaveworks/eksctl/blob/79373be4ffc37ae59e8822232a7899c2c8d4efd8/pkg/gitops/flux/wait.go#L21-L102

I understand that the first instance was to get SSH key that Flux generates, which arguably could had been obtained by using only Kubernetes client and waiting on the secret, which is a separate issue (https://github.com/weaveworks/eksctl/issues/1213).

---

_Part of: #1180_

**Comment by cPu1** [comment]
Both Flux and the Helm Operator deployments define a readiness probe for their app containers, so we don't need to access the container port to determine if they're ready. Instead of port-forwarding, we could watch the pod status using the Kubernetes Watch API, as the API server handles transitioning the pod status to `Ready` once the readiness probe succeeds. This is also arguably better (and probably faster if the polling time is high enough) than polling the container's port. 

>I understand that the first instance was to get SSH key that Flux generates, which arguably could had been obtained by using only Kubernetes client and waiting on the secret, which is a separate issue (#1213).

Even if we continue to obtain the SSH key via port-forwarding, we should use the Watch API to watch for pod readiness before forwarding to the container port.

For the Helm Operator, no port forwarding should be required as it requests the same `/healthz` endpoint that's configured as a readiness probe for its app container.
